### PR TITLE
initialize_matrix python to perl with Box-Muller

### PR DESCRIPTION
**Pull request by sas91**
Added a perl version of "initialize_matrix" script that generates uniform random numbers in pairs and then uses Box-Muller to make them Gaussian. This gives approximately 3.5x speed up compared to the python script.

### CODE REVIEW

File: scripts/rnnlm/initialize_matrix.pl [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env perl
+# Copyright 2017     Hossein Hadian
+
+$randseed = 0;
+$shifted = 0;
+
+do {
+  $shifted=0;
+  if ($ARGV[0] eq "--first-column") {
```
Comment (danpovey) [inline scripts/rnnlm/initialize_matrix.pl]: thanks!  Can you please add the --first-element option that now exists in the source python script, replace all invocations of the python script with invocations of the perl script, and delete the python script?  Then I'll merge.

Comment (sas91) [inline scripts/rnnlm/initialize_matrix.pl]: Thank you Dr. Povey. Added --first-element option, replaced all invocations of the python script with invocations of the perl script, and the python script is deleted.

File: egs/ptb/s5/local/rnnlm/train_rnnlm_sparse.sh [hunk 2]
```diff
@@ -39,19 +39,11 @@ EOF
 steps/nnet3/xconfig_to_configs.py --xconfig-file $dir/configs/network.xconfig --config-dir $dir/configs/
 nnet3-init $dir/configs/final.config - | nnet3-copy --learning-rate=0.0001 - $dir/0.rnnlm
 
-# note: this is way too slow, we need to speed it up somehow.
-# I'm not sure if I want to have a dependency on numpy just for this though.
-# maybe we can rewrite in perl.
 # we may later initialize this to identity for the first block.
-rnnlm/initialize_matrix.py --num-rows=$feat_dim --num-cols=$embedding_dim \
-                           --first-element=1.0 --stddev=0.001 > $dir/embedding.0.mat
+perl rnnlm/initialize_matrix.pl --first-element 1.0 --stddev 0.001 $feat_dim $embedding_dim > $dir/embedding.0.mat
```
Comment (danpovey) [inline egs/ptb/s5/local/rnnlm/train_rnnlm_sparse.sh]: can you please just invoke as rnnlm/initialize_matrix.pl ?
Make sure execute permission is set in the git repo and you have the shebang there.

File: scripts/rnnlm/initialize_matrix.pl [hunk 3]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env perl
+# Copyright 2017     Hossein Hadian
```
Comment (danpovey) [inline scripts/rnnlm/initialize_matrix.pl]: and you can add your own name here.

File: scripts/rnnlm/initialize_matrix.pl [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
```
Comment (danpovey) [inline scripts/rnnlm/initialize_matrix.pl]: as far as I can see from github, you didn't set the mode right.  there is a special invocation for when you forgot to give it the right mode before git add-ing it.  look it up.

File: scripts/rnnlm/initialize_matrix.pl (hunk 3 above)
Comment (sas91) [inline scripts/rnnlm/initialize_matrix.pl]: I have added now.

File: egs/ptb/s5/local/rnnlm/train_rnnlm_sparse.sh (hunk 2 above)
Comment (sas91) [inline egs/ptb/s5/local/rnnlm/train_rnnlm_sparse.sh]: Yes I have changed the way it is invoked. I gave it execute permission in the grid and then pushed it to the git repo.

File: scripts/rnnlm/initialize_matrix.pl (hunk 3 above)
Comment (danpovey) [inline scripts/rnnlm/initialize_matrix.pl]: you didn't push

**Pull request merged**